### PR TITLE
CC-63-Sidebar-Refactoring

### PR DESCRIPTION
**Pull request by 17Sergey**

### CODE REVIEW

**Comment by 17Sergey** [comment]
Guys I have a problem. In Firefox, Edge and Opera everything is ok, but in Chrome there comes a bug with Decoration in Preview Block.
Normal:
image

Bug: (different border widths)
image

I don't know why but could you check this deploy preview in your browsers?

**Comment by PavelPyasecky** [comment]
> Guys I have a problem. In Firefox, Edge and Opera everything is ok, but in Chrome there comes a bug with Decoration in Preview Block. Normal: image
> 
> Bug: (different border widths) image
> 
> I don't know why but could you check this deploy preview in your browsers?

Let's leave it as is. There is separate task for reengineering this label panel. I'm able to reproduce the bug in Chrome and Firefox browsers.

**Comment by PavelPyasecky** [comment]
I've found one thing. The current sidebar implementation is a bit bigger and wider than previous one by 10px. I believe that's because the current Logo is wider by 10px. Guys, do we want such innovations? @17Sergey @ArtsemiLasyi   
Merged_document_2

And the initial state of the arrow has been reversed. 
image

File: src/components/Sidebar/SidebarHeader/SidebarHeader.tsx [hunk 1]
```diff
@@ -1,20 +1,56 @@
...
+const StyledHeaderGrid = styled(StyledSidebarGrid) <{ $sidebarExpanded?: boolean; }>`
+    padding-top: 0;
+    padding-bottom: 0;
+`
+
+const StyledTitle = styled.div<{ $sidebarExpanded: boolean }>`
+    /* These properties make the width of title text animate while collapsing! 
+    Great solution with grid */
+    display: grid;
+    grid-template-columns: ${props => props.$sidebarExpanded ? "1fr" : "0fr"};
```
Comment (PavelPyasecky) [inline src/components/Sidebar/SidebarHeader/SidebarHeader.tsx:23]: Good point!

File: src/styles/Global.ts [hunk 2]
```diff
@@ -4,6 +4,18 @@ import fontsCss from './fonts.module.css'
...
+:root {
+  font-size: 16px;
+  /* Sidebar variables */
+  --sidebar-grid-left-col: 3em;
+  --sidebar-grid-gap: 0.5em;
+  --sidebar-grid-padding: 1em;
+
+  /* Content */
+  --content-margin:
+  calc(var(--sidebar-grid-left-col) + var(--sidebar-grid-gap) + var(--sidebar-grid-padding) * 2);
```
Comment (PavelPyasecky) [inline src/styles/Global.ts:16]: Nice solution!

**Comment by 17Sergey** [comment]
@PavelPyasecky the arrow reversed is neccessary. I found this bug, when user clicks on the arrow, another(reversed) icon starts loading and user sees like it has disappeared. So we use only one icon, just animate it to reverse

**Comment by ArtsemiLasyi** [comment]
> I've found one thing. The current sidebar implementation is a bit bigger and wider than previous one by 10px. I believe that's because the current Logo is wider by 10px. Guys, do we want such innovations? @17Sergey @ArtsemiLasyi Merged_document_2
> 
> And the initial state of the arrow has been reversed. image

@17Sergey, I agree with @PavelPyasecky. Current sidebar behavior is not expected by user. Let's have a call if you want

**Comment by ArtsemiLasyi** [comment]
@17Sergey, I am not sure that current expand icon view is correct. Seems like it reversed. I don;t think that it is expected behaviour
image

File: src/components/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -1,42 +1,49 @@
...
+const StyledSidebar = styled.aside`
+    position: fixed;
+    top: 0;
+    left: 0;
+    z-index: 10;
 
-type SidebarProps = {};
+    /* SVH is a new CSS unit and not all browsers use it yet */
+    height: 100vh;
+    height: 100svh;
```
Comment (ArtsemiLasyi) [inline src/components/Sidebar/Sidebar.tsx]: Do we really need SVH here? Why?

File: src/components/Sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -1,42 +1,49 @@
...
+    /* SVH is a new CSS unit and not all browsers use it yet */
+    height: 100vh;
+    height: 100svh;     
+
+    background-color: #fff;
+    color: rgb(75 85 99);
 
-const StyledAside = styled.aside`
-    background: ${({ theme }) => theme.colors.background};
+    border-right: 1px solid #e5e7eb;
```
Comment (ArtsemiLasyi) [inline src/components/Sidebar/Sidebar.tsx]: Let's try to keep all colors in separate file or in theme variable

Comment (ArtsemiLasyi) [reply src/components/Sidebar/Sidebar.tsx]: This also applies to all other similar places

File: src/components/Sidebar/SidebarHeader/SidebarHeader.tsx [hunk 5]
```diff
@@ -1,20 +1,56 @@
...
-        <div className={`flex items-center ${expanded ? "" : "justify-center"}`}>
-            <ChashaIcon width={68} height={92} />
-            <span
-                className={`overflow-hidden font-bold text-gray-600 transition-all mt-2 ${expanded ? "" : "w-0 ml-0"}`}>
-                Неупиваемая Чаша
-            </span>
-        </div>
+        <StyledHeaderContent>
+            <StyledHeaderGrid>
+                <ChashaIcon width={"3em"} />
```
Comment (ArtsemiLasyi) [inline src/components/Sidebar/SidebarHeader/SidebarHeader.tsx:48]: Can we use theme variable here?

Comment (17Sergey) [reply src/components/Sidebar/SidebarHeader/SidebarHeader.tsx:48]: To what?

**Review by ArtsemiLasyi — APPROVED** [review]

**Review by PavelPyasecky — APPROVED** [review]

**Pull request merged**